Description: Upstream changes introduced in version 2.1.0-1~maverickubuntu5
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 bumblebee (2.1.0-1~maverickubuntu5) maverick; urgency=low
 .
   * Based on git-changes v. 1.6.68..
 .
 The person named in the Author field signed this changelog entry.
Author: Martin Juhl (MrMEEE) <mj@casalogic.dk>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- bumblebee-2.1.0.orig/install-files/bumblebee-configuration
+++ bumblebee-2.1.0/install-files/bumblebee-configuration
@@ -53,10 +53,10 @@ if [ "$UID" != "$ROOT_UID" ] ; then
 	exit 1
 fi
   
-#Determine Arch x86_64 or i686
+#	Determine Arch x86_64 or i686
 ARCH=`uname -m`
 
-#Get tools location 
+#	Get tools location 
 LSPCI=`which lspci` 
 MODPROBE=`which modprobe`
 
@@ -75,18 +75,24 @@ else
 	rm $DEFAULT_FILE
 fi
 
-if [ -f /tmp/machineinfo ]; then   
+if [ -f /tmp/machineinfo ]; then
 	rm /tmp/machineinfo
 fi
 
 if [ "$DISTRO" == "UBUNTU" ] ; then
-	update-alternatives --remove gl_conf /usr/lib/nvidia-current/ld.so.conf
+	for arch in x86_64-linux-gnu i386-linux-gnu; do
+		if update-alternatives --list ${arch}_gl_conf >/dev/null 2>&1; then
+			update-alternatives --force --set ${arch}_gl_conf /usr/lib/$arch/mesa/ld.so.conf
+		fi
+	done
+	if update-alternatives --list gl_conf >/dev/null 2>&1; then
+		update-alternatives --force --set gl_conf /usr/lib/mesa/ld.so.conf
+	fi
 	rm -rf /etc/alternatives/xorg_extra_modules
 	rm -rf /etc/alternatives/xorg_extra_modules-bumblebee
 	rm -rf /usr/lib/xorg/x11-extra-modules/xorg
 	rm -rf /usr/lib/xorg/extra-modules
 	rm -rf /usr/lib/nvidia-current/xorg/xorg
-	update-alternatives --remove x86_64-linux-gnu_gl_conf /usr/lib/nvidia-current/ld.so.conf
 	ln -s /usr/lib/nvidia-current/xorg /etc/alternatives/xorg_extra_modules-bumblebee
 	ldconfig
 fi
@@ -164,12 +170,12 @@ while [ "$CONFIGMODE" != "MANUAL" ] && [
 			else
 				CONFIGMODE=MANUAL
 			fi
-			;;   
+			;;
 		1)
-			CONFIGMODE=AUTO   
+			CONFIGMODE=AUTO
 			MANUFACTURER=`cat /tmp/machineinfo |cut -f1 -d\|`
 			MODEL=`cat /tmp/machineinfo |cut -f2 -d\|`
-			USER=`cat /tmp/machineinfo |cut -f9 -d\|` 
+			USER=`cat /tmp/machineinfo |cut -f9 -d\|`
 			CONNECTEDMONITOR=`cat /tmp/machineinfo |cut -f3 -d\| | sed -e 's/ //g'`
 			INTELBUSID=`cat /tmp/machineinfo |cut -f4 -d\| | sed -e 's/ //g'`
 			NVIDIABUSID=`cat /tmp/machineinfo |cut -f5 -d\| | sed -e 's/ //g'`
@@ -178,7 +184,7 @@ while [ "$CONFIGMODE" != "MANUAL" ] && [
 			OS=`cat /tmp/machineinfo |cut -f10 -d\|`
 			POWERSTATUS=`cat /tmp/machineinfo |sed -n 1p|cut -f6 -d\| | sed -e 's/ //g'`
 			;;
-		*) 
+		*)
 			LIST_SIZE=$(($LINES - 10))
 			choice=$(whiptail --nocancel --menu \
 " Please select a machine profile to install.\n\n"\
@@ -205,7 +211,7 @@ while [ "$CONFIGMODE" != "MANUAL" ] && [
 				OS=`cat /tmp/machineinfo |sed -n \`echo $choice\`p|cut -f10 -d\|`
 				POWERSTATUS=`cat /tmp/machineinfo |sed -n \`echo $choice\`p|cut -f6 -d\| | sed -e 's/ //g'`
 			fi
-			;; 
+			;;
 		esac
 	fi
 
@@ -279,15 +285,26 @@ if [ "$CONFIGMODE" == "MANUAL" ]; then
 	elif [ `${LSPCI} |grep 3D |wc -l` -eq 1 ]; then
 		NVIDIABUSID=`${LSPCI} | grep 3D  | grep nVidia | awk '{ print "PCI:"$1 }' | sed s/\\\./:/g`
 	else
-		whiptail --msgbox \
+		if [ "$DISTRO" == "OPENSUSE" ] ; then
+			whiptail --msgbox \
+" The BusID of the nVidia card can't be determined.\n"\
+"You must correct this manually in /etc/X11/xorg.conf.nvidia and /etc/X11/xorg.conf.nouveau\n"\
+"Please report this problem..\n\n"\
+"Press Any Key to continue." $LINES $COLUMNS
+		else
+			whiptail --msgbox \
 " The BusID of the nVidia card can't be determined.\n"\
-"You must correct this manually in /etc/X11/xorg.conf.nvidia.\n"\
+"You must correct this manually in /etc/X11/xorg.conf.nvidia\n"\
 "Please report this problem..\n\n"\
 "Press Any Key to continue." $LINES $COLUMNS
+		fi
 	fi
 fi
 
 sed -i "s/\([ \t]*BusID[ \t]*\)\".*\"/\1\"$NVIDIABUSID\"/g" /etc/X11/xorg.conf.nvidia
+if [ "$DISTRO" == "OPENSUSE" ] ; then
+	sed -i "s/\([ \t]*BusID[ \t]*\)\".*\"/\1\"$NVIDIABUSID\"/g" /etc/X11/xorg.conf.nouveau
+fi
 
 if [ "$CONFIGMODE" == "MANUAL" ]; then
 	CONNECTEDMONITOR="UNKNOWN"
@@ -299,7 +316,7 @@ if [ "$CONFIGMODE" == "MANUAL" ]; then
 	fi
 	if [ "$DISTRO" == "UBUNTU" ] ; then
 		${MODPROBE} nvidia-current
-		if [ "`lspci -v -s 01:00.0 |grep "Unknown header type"`" == "" ]; then
+		if [ "`lspci -v -s 01:00.0 | grep "Unknown header type"`" == "" ] ; then
 			if [ `LD_LIBRARY_PATH=/usr/lib/nvidia-current /usr/lib/nvidia-current/bin/nvidia-xconfig --query-gpu-info | grep "Display Devices" | cut -f2 -d":"` -gt 0 ] ; then
 				CONNECTEDMONITOR=`LD_LIBRARY_PATH=/usr/lib/nvidia-current /usr/lib/nvidia-current/bin/nvidia-xconfig --query-gpu-info | grep "Display Device 0" | cut -f2 -d\( | cut -f1 -d\)`
 			fi
@@ -308,7 +325,6 @@ if [ "$CONFIGMODE" == "MANUAL" ]; then
 	fi
 	if [ "$DISTRO" == "OPENSUSE" ] ; then
 		${MODPROBE} nvidia 2>&1 >/dev/null
-
 		if [ -x /usr/bin/nvidia-xconfig ] ; then
 			if [ "`/usr/bin/nvidia-xconfig --query-gpu-info 2>&1 | grep ERROR`" == "" ] ; then
 				if [ `/usr/bin/nvidia-xconfig --query-gpu-info | grep "Display Devices" | cut -f2 -d":"` -gt 0 ] ; then
@@ -332,16 +348,16 @@ choice=$(whiptail --nocancel --menu \
 "manually choose the device, if your having problems, try another." $LINES $COLUMNS $LIST_SIZE 1 "CRT-0" 2 "DFP-0" 3 "Manually Enter" 3>&1 1>&2 2>&3)
 		case "$choice" in
 		1)
-		      CONNECTEDMONITOR="CRT-0"
+			CONNECTEDMONITOR="CRT-0"
 		;;
 		2)
-		      CONNECTEDMONITOR="DFP-0"
+			CONNECTEDMONITOR="DFP-0"
 		;;
 		3)
-CONNECTEDMONITOR=$(whiptail --inputbox " Enter the nVidia connected monitor:" $LINES $COLUMNS 3>&1 1>&2 2>&3)
+			CONNECTEDMONITOR=$(whiptail --inputbox " Enter the nVidia connected monitor:" $LINES $COLUMNS 3>&1 1>&2 2>&3)
 		;;
 		esac
-fi
+	fi
 fi
 
 sed -i "s/\([ \t]*Option[ \t]*\"ConnectedMonitor\"[ \t]*\)".*"/\1\"$CONNECTEDMONITOR\"/g" /etc/X11/xorg.conf.nvidia
@@ -349,10 +365,10 @@ sed -i "s/\([ \t]*Option[ \t]*\"Connecte
 #
 #	Always try to enable the card before switching scripts
 #
-if [ -f /usr/local/bin/bumblebee-enablecard ]; then
+if [ -f /usr/local/bin/bumblebee-enablecard ] ; then
 	/usr/local/bin/bumblebee-enablecard
 fi
-if [ -f /usr/bin/bumblebee-enablecard ]; then
+if [ -f /usr/bin/bumblebee-enablecard ] ; then
 	/usr/bin/bumblebee-enablecard
 fi
 if [ "$DISTRO" == "UBUNTU" ] ; then
@@ -394,93 +410,61 @@ if [ "$DISTRO" == "OPENSUSE" ] ; then
 	#
 	#	Setup new scripts
 	#
-	if [ ! "$POWERON" == "UNAVAILABLE" ] ; then
-		if [ -f /usr/local/bin/optirun ] ; then
-			#
-			#	Script install
-			#
-			echo $POWERON | xxd -r -p > /usr/local/bin/bumblebee-enablecard
-			echo $POWEROFF | xxd -r -p > /usr/local/bin/bumblebee-disablecard
-		elif [ -f /usr/bin/optirun ] ; then
-			#
-			#	Repository install
-			#
-			echo $POWERON | xxd -r -p > /usr/bin/bumblebee-enablecard
-			echo $POWEROFF | xxd -r -p > /usr/bin/bumblebee-disablecard
-		fi
+	if [ -f /usr/local/bin/optirun ] ; then
+		SCRIPT_PATH=/usr/local/bin
+	elif [ -f /usr/bin/optirun ] ; then
+		SCRIPT_PATH=/usr/bin
 	else
-		if [ -f /usr/local/bin/optirun ] ; then
+		SCRIPT_PATH=""
+	fi
+	if [ "$SCRIPT_PATH" != "" ] ; then
+		if [ ! "$POWERON" == "UNAVAILABLE" ] ; then
 			#
-			#	Script install
+			#	Power scripts found
 			#
-			whiptail --msgbox \
-"No power-on/off configuration has been reported for your machine,\n"\
-"please manually enter configuration in:\n\n"\
-"/usr/local/bin/bumblebee-enablecard and /usr/local/bin/bumblebee-disablecard\n\n"\
-"Some examples are available in /usr/share/doc/packages/bumblebee/examples/.\n\n"\
-"Press any key to continue..." $LINES $COLUMNS
-			if [ ! -f /usr/local/bin/bumblebee-enablecard ] ; then
-				# Not installed
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
-			else
-				# Already Exists
-				echo
-				echo "nVidia card enable-script: /usr/local/bin/bumblebee-enablecard exists moving old script to /usr/local/bin/bumblebee-enablecard.old."
-				echo
-				mv /usr/local/bin/bumblebee-enablecard /usr/local/bin/bumblebee-enablecard.old
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
-			fi
-			sed -i "s?INSERTPOWERON?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/local/bin/bumblebee-enablecard
+			echo $POWERON | xxd -r -p > $SCRIPT_PATH/bumblebee-enablecard
+			echo $POWEROFF | xxd -r -p > $SCRIPT_PATH/bumblebee-disablecard
 
-			if [ ! -f /usr/local/bin/bumblebee-disablecard ] ; then
-				# Not installed
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
-			else
-				# Already Exists
-				echo
-				echo "nVidia card disable-script: /usr/local/bin/bumblebee-disablecard exists, moving old script to /usr/local/bin/bumblebee-disablecard.old."
-				echo
-				mv /usr/local/bin/bumblebee-disablecard /usr/local/bin/bumblebee-disablecard.old
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
-			fi
-			sed -i "s?INSERTPOWEROFF?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/local/bin/bumblebee-disablecard
-			chmod +x /usr/local/bin/bumblebee*
-		elif [ -f /usr/bin/optirun ] ; then
+			sed -i 's?modprobe nvidia?if \[ -f /usr/bin/nvidia-xconfig \] ; then\n\tmodprobe nvidia\nelse\n\tmodprobe nouveau\nfi?g' $SCRIPT_PATH/bumblebee-enablecard
+			sed -i 's?modprobe -r nvidia?modprobe -r nouveau >/dev/null 2>\&1\nif lsmod | grep -q nouveau; then\n\techo \"Error: could not unload nouveau module, leaving card turned on\"\n\texit\nfi\nmodprobe -r nvidia >/dev/null 2>\&1?g' $SCRIPT_PATH/bumblebee-disablecard
+		else
 			#
-			#	Repository install
+			#	No power script found
 			#
 			whiptail --msgbox \
 "No power-on/off configuration has been reported for your machine,\n"\
 "please manually enter configuration in:\n\n"\
-"/usr/bin/bumblebee-enablecard and /usr/bin/bumblebee-disablecard\n\n"\
+"$SCRIPT_PATH/bumblebee-enablecard and $SCRIPT_PATH/bumblebee-disablecard\n\n"\
 "Some examples are available in /usr/share/doc/packages/bumblebee/examples/.\n\n"\
 "Press any key to continue..." $LINES $COLUMNS
-			if [ ! -f /usr/bin/bumblebee-enablecard ] ; then
+			if [ ! -f $SCRIPT_PATH/bumblebee-enablecard ] ; then
 				# Not installed
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/bin/bumblebee-enablecard
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template $SCRIPT_PATH/bumblebee-enablecard
 			else
 				# Already Exists
 				echo
-				echo "nVidia card enable-script: /usr/bin/bumblebee-enablecard exists moving old script to /usr/bin/bumblebee-enablecard.old."
+				echo "nVidia card enable-script: $SCRIPT_PATH/bumblebee-enablecard exists moving old script to $SCRIPT_PATH/bumblebee-enablecard.old."
 				echo
-				mv /usr/bin/bumblebee-enablecard /usr/bin/bumblebee-enablecard.old
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/bin/bumblebee-enablecard
+				mv $SCRIPT_PATH/bumblebee-enablecard $SCRIPT_PATH/bumblebee-enablecard.old
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template $SCRIPT_PATH/bumblebee-enablecard
 			fi
-			sed -i "s?INSERTPOWERON?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/bin/bumblebee-enablecard
+			sed -i "s?INSERTPOWERON?echo \"Please run bumblebee-configuration to enable power management.\"?g" $SCRIPT_PATH/bumblebee-enablecard
+			sed -i 's?modprobe nvidia?if \[ -f /usr/bin/nvidia-xconfig \] ; then\n\tmodprobe nvidia\nelse\n\tmodprobe nouveau\nfi?g' $SCRIPT_PATH/bumblebee-enablecard
 
-			if [ ! -f /usr/bin/bumblebee-disablecard ] ; then
+			if [ ! -f $SCRIPT_PATH/bumblebee-disablecard ] ; then
 				# Not installed
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/bin/bumblebee-disablecard
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template $SCRIPT_PATH/bumblebee-disablecard
 			else
 				# Already Exists
 				echo
-				echo "nVidia card disable-script: /usr/bin/bumblebee-disablecard exists, moving old script to /usr/bin/bumblebee-disablecard.old."
+				echo "nVidia card disable-script: $SCRIPT_PATH/bumblebee-disablecard exists, moving old script to $SCRIPT_PATH/bumblebee-disablecard.old."
 				echo
-				mv /usr/bin/bumblebee-disablecard /usr/bin/bumblebee-disablecard.old
-				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/bin/bumblebee-disablecard
+				mv $SCRIPT_PATH/bumblebee-disablecard $SCRIPT_PATH/bumblebee-disablecard.old
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template $SCRIPT_PATH/bumblebee-disablecard
 			fi
-			sed -i "s?INSERTPOWEROFF?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/bin/bumblebee-disablecard
-			chmod +x /usr/bin/bumblebee*
+			sed -i "s?INSERTPOWEROFF?echo \"Please run bumblebee-configuration to enable power management.\"?g" $SCRIPT_PATH/bumblebee-disablecard
+			sed -i 's?modprobe -r nvidia?modprobe -r nouveau >/dev/null 2>\&1\nif lsmod | grep -q nouveau; then\n\techo \"Error: could not unload nouveau module, leaving card turned on\"\n\texit\nfi\nmodprobe -r nvidia >/dev/null 2>\&1?g' $SCRIPT_PATH/bumblebee-disablecard
+			chmod +x $SCRIPT_PATH/bumblebee*
 		fi
 	fi
 fi
@@ -519,9 +503,9 @@ case "$choice" in
 	IMAGETRANSPORT="rgb"
 	;;
 *)
-	IMAGETRANSPORT="xv"  
+	IMAGETRANSPORT="xv"
 	;;
-esac 
+esac
 
 sed -i 's/VGL_COMPRESS.*/VGL_COMPRESS='$IMAGETRANSPORT'/' $DEFAULT_FILE
 
